### ISSUE TITLE
Couldn't load SIP module. Python support will be disabled.

### ISSUE THREAD

**Issue by kulpojke** [qgis/QGIS#50591]
### What is the bug or the crash?

Upon Starting QGIS recieve the following error:

```
Couldn't load SIP module.
Python support will be disabled.


Traceback (most recent call last):
  File "", line 1, in 
ModuleNotFoundError: No module named 'qgis'


Python version:
3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110]

QGIS version:
3.26.3-Buenos Aires 'Buenos Aires', 65e4edfdad

Python path:
['/usr/share/qgis/python', '/home/michael/.local/share/QGIS/QGIS3/profiles/default/python', '/home/michael/.local/share/QGIS/QGIS3/profiles/default/python/plugins', '/usr/share/qgis/python/plugins', '/home/michael', '/usr/share/qgis/python', '/home/michael/.local/share/QGIS/QGIS3/profiles/default/python', '/home/michael/.local/share/QGIS/QGIS3/profiles/default/python/plugins', '/usr/share/qgis/python/plugins', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/usr/lib/python3/dist-package', '/home/michael/miniconda3/lib/python39.zip', '/home/michael/miniconda3/lib/python3.9', '/home/michael/miniconda3/lib/python3.9/lib-dynload', '/home/michael/.local/lib/python3.9/site-packages', '/home/michael/miniconda3/lib/python3.9/site-packages']

```

### Steps to reproduce the issue

Start QGIS from terminal or from application launcher.

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.26.3-Buenos Aires | QGIS code revision | 65e4edfdad
-- | -- | -- | --
Qt version | 5.15.2
Python version | 3.9.2
GDAL/OGR version | 3.2.2
PROJ version | 7.2.1
EPSG Registry database version | v10.008 (2020-12-16)
GEOS version | 3.9.0-CAPI-1.16.2
SQLite version | 3.34.1
PDAL version | 2.2.0
PostgreSQL client version | 13.8 (Debian 13.8-0+deb11u1)
SpatiaLite version | 5.0.1
QWT version | 6.1.4
QScintilla2 version | 2.11.6
OS version | Debian GNU/Linux 11 (bullseye)
  |   |   |  

</body></html>QGIS version
3.26.3-Buenos Aires
QGIS code revision
[65e4edfdad](https://github.com/qgis/QGIS/commit/65e4edfdad)
Qt version
5.15.2
Python version
3.9.2
GDAL/OGR version
3.2.2
PROJ version
7.2.1
EPSG Registry database version
v10.008 (2020-12-16)
GEOS version
3.9.0-CAPI-1.16.2
SQLite version
3.34.1
PDAL version
2.2.0
PostgreSQL client version
13.8 (Debian 13.8-0+deb11u1)
SpatiaLite version
5.0.1
QWT version
6.1.4
QScintilla2 version
2.11.6
OS version
Debian GNU/Linux 11 (bullseye)

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

I have been using 3.26 for a few weeks and everything has been fine. Yesterday morning when I turned on my computer this was happening.  The evening before Debian performed a software update.

If I run the Python QGIS seems to be running based on the error, and import the qgis module it works fine.
```
x:/usr/bin$ python3.9
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import qgis
>>> qgis
<module 'qgis' from '/usr/lib/python3/dist-packages/qgis/__init__.py'>
```

Python 3.9.2 is not my system default Python. Annoyingly the default  is Anaconda,
```
x:/usr/bin$ which python
/home/michael/miniconda3/envs/pdalpy/bin/python
```
 The qgis module is not found in the anaconda Python distribution.

```
x:/usr/bin$ python
Python 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:35:26) [GCC 10.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import qgis
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'qgis'
>>> 
```
But, according to the error message QGIS is not using the anaconda python.  

As a test I exported /usr/bin to the front of my PATH (using export statement towards the end of  `/usr/bin/qgis`) The error goes away, but it brakes a whole bunch of other things (also I noted it was looking inside of an anaconda env directory for proj_4 after I made this change.)  I then apt purged QGIS and reinstalled to undo my path change. 

I have also tried uninstalling pyqt5, as that was a solution offered in past posts with issues similar to this,
```
sudo apt-get remove  python3-pyqt5
```
This seems to uninstall QGIS, so I installed QGIS again, and the problem persists.

Thanks

**Comment by nicogodet** [comment]
It's most likely a local issue due to your setup rather than a QGIS issue unfortunately...

Could you try with a new Unix user ?

**Comment by kulpojke** [comment]
After much tinkering I fixed it. As mentioned earlier python 3.9 will import qgis
```
x:/usr/bin$ python3.9
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import qgis
>>> qgis
<module 'qgis' from '/usr/lib/python3/dist-packages/qgis/__init__.py'>
```
In QGIS, I went  to Settings > Options >System >Environment. I selected the 'Use custom variables' checkbox and added the variable PYTHONPATH (which was not set in the variables listed),  I set the PYTHONPATH to the location of qgis in the last line of the above code block, `/usr/lib/python3/dist-packages`.  Then I restarted QGIS and it worked.

So, as @[nicogodet](https://github.com/nicogodet) pointed out, it was a local setup issue.  It seems like variations on this problem are common for conda users.  Hopellly this will be helpful for others.

**Comment by blxl909** [comment]
I have this problem too, and I keep debugging this for a whole day, I found that the proposed solutions did not solve my problem, so I will write down my own solution. First of all, I am a non-root user, and the what cause this issue is actually the system not having the correct Python path set. Therefore, our goal is to find this Python path. I discovered that entering QGIS in non-command line mode does not cause this problem, and the Python console is also usable. So, the solution becomes simple: I just need to enter the Python console in the plugins menu and execute the following command:
`import sys
print(sys.path)
import qgis
print(qgis.__file__)
import sip
print(sipqgis.__file__)`

you will find that the path is not in sys.path
so,we `export PYTHONPATH="/usr/lib/python3/dist-packages:$PYTHONPATH"`(this is just an example, use your own case)
then the problem solved.

(my english is not good , so I use chatgpt to generate some sentences)
